oar/core/advisory.py

```diff
@@ -629,3 +662,103 @@ def has_blocking_secruity_alert(self):
             return blocking
         else:
             return False
+        
+    def get_unhealthy_builds(self):
+        """
+        Get unhealthy builds of advisory.
+
+        Returns:
+            list: Unhealthy builds.
+        """
+        nvrs = self.get_build_nvrs()
+
+        unhealthy_builds = []
+
+        for n in nvrs:
+            build_grades = self.get_build_grades(n)
+            for bg in build_grades:
+                if not util.is_grade_healthy(bg["grade"]):
+                    unhealthy_builds.append(bg)
+
+        return unhealthy_builds
+        
+    def get_build_grades(self, nvr):
+        """
+        Get all architecture grades of build.
+
+        Returns:
+            list: All architecture grades of build {nvr, architecture, grade}.
+        """
+        nvr_url = "https://pyxis.engineering.redhat.com/v1/images/nvr/%s?include=data.freshness_grades&include=data.architecture" % nvr
+        resp = requests.get(nvr_url, auth=self._auth, verify=self.ssl_verify)
+
+        nvr_grades = []
+
+        if resp.ok:
+            # TODO should we skip empty data? 
+
+            for arch in resp.json()["data"]:
+                effective_grade = None
+                for fg in arch["freshness_grades"]:                    
+                    checked_grade = fg["grade"]
+                    start_date = parser.parse(fg["start_date"])
+
+                    # skip if checked grade is not effective yet
+                    if start_date > datetime.now(timezone.utc):
+                        continue
+
+                    # skip if new_grade is 
+                    if "end_date" in fg and parser.parse(fg["end_date"]) < datetime.now(timezone.utc):
+                        continue
+
+                    # skip if checked grade is smaller than current effective grade(if intervals are overlapping, bigger grade is taken)
+                    if effective_grade and checked_grade < effective_grade:
+                        continue
+
+                    # save new effective grade
+                    effective_grade = checked_grade
+
+                # TODO create object instead dict?
```

Comment:
@rioliu-rh Is it ok to return a dict? Or should I transform this to an object or json?

Reply:
you mean create a class to present grade, put the jsondata as input of constructor?

Reply:
yes, I mean from implementation detail point of view if t is ok to keep it like this or I should return some more sophisticated structure like json or object.

Reply:
my opinion is: we don't have to, grade info is only used for this kinda of checking, it won't be used across modules.